### PR TITLE
Add deprecated error when using 'pattern' option.

### PR DESCRIPTION
**Pull request by IndraGunawan**

### CODE REVIEW

File: Routing/Loader/RestYamlCollectionLoader.php [hunk 1]
```diff
@@ -112,6 +112,8 @@ public function load($file, $type = null)
                 // the path option
                 if (!isset($config['path'])) {
                     $config['path'] = $config['pattern'];
+
+                    @trigger_error('The "pattern" option is deprecated. Use the "path" option instead.', E_USER_DEPRECATED);
```
Comment (Ener-Getick) [inline Routing/Loader/RestYamlCollectionLoader.php]: Maybe include the file name ?

Comment (IndraGunawan) [inline Routing/Loader/RestYamlCollectionLoader.php]: done :D

File: Routing/Loader/RestYamlCollectionLoader.php [hunk 2]
```diff
@@ -112,6 +112,8 @@ public function load($file, $type = null)
                 // the path option
                 if (!isset($config['path'])) {
                     $config['path'] = $config['pattern'];
+
+                    @trigger_error(sprintf('The "pattern" option in "%s" is deprecated. Use the "path" option instead.', $path), E_USER_DEPRECATED);
```
Comment (Ener-Getick) [inline Routing/Loader/RestYamlCollectionLoader.php]: And what about the route name ? (Sorry i didn't think about this one earlier)

**Comment by Ener-Getick** [comment]
Sorry one more thing... Could you add it to the UPGRADING files?

**Comment by Ener-Getick** [comment]
Thank you @IndraGunawan

**Pull request merged**